image/verify/src/verifier.rs

```diff
@@ -595,6 +609,21 @@ impl<Env: ImageVerificationEnv> ImageVerifier<Env> {
             .sha384_digest(image, range.start, range.len() as u32)
             .map_err(|_| err_u32!(VendorPubKeyDigestFailure))
     }
+
+    /// Calculates the effective minimum SVN.
+    ///
+    /// If anti-rollback is disabled, the effective min-SVN is zero.
+    /// Otherwise, it is MAX(min-SVN, SVN-fuses), because SVN-fuses will
+    /// be updated to match min-SVN.
```

Comment:
Is this the same logic that BootROM will do for image authentication?

Reply:
I would expect that boot ROM would not have to do the `MAX(min-SVN, SVN-fuses)` bit when authenticating the image.

This is where some feedback would be helpful. I went with MAX here because without it, if we just sample SVN-fuses, then on the first boot after min-SVN revs those fuses the DICE evidence may reflect what the fuses _were_, not what they are now (this assuming that SVN-fuses get blown after Caliptra boots, not before). A subsequent power cycle would cause the evidence to catch up to the fuse state.

Reply:
There's multiple layers here because what Caliptra sees are registers and not the fuses themselves. The registers are latched on Caliptra power-good and locked as part of the Boot FSM transition states. So there's no way for Caliptra ROM to know any fuse state change until the next power-good anyway.

Either ROM or FW acting as update authorizer needs to do the `MAX(min-SVN, SVN-fuses)` logic to handle hitless update cases, because min-SVN can rev on update reset via manifest, but SVN-fuses cannot. I prefer ROM does it so we're not implementing authorization logic in different places.

Reply:
I see; it sounds like in either case, we should be writing this effective-min-SVN out to the DataVault.

Should we try and create a mechanism for Caliptra ROM to signal to SoC what the new value of SVN-fuses should be?

Also, I gather there's an assumption here that upon hitless update, while we can increment some "soft" min-SVN and stash it in the DataVault, we can't increment the "hard" min-SVN stored in fuses?

Reply:
> Should we try and create a mechanism for Caliptra ROM to signal to SoC what the new value of SVN-fuses should be?

That can be useful, but it's also possible for the SoC to just manage the fuses orthogonally from what Caliptra says.

> Also, I gather there's an assumption here that upon hitless update, while we can increment some "soft" min-SVN and stash it in the DataVault, we can't increment the "hard" min-SVN stored in fuses?

Yep. @vsonims @bharatpillilli for vis

Reply:
Ack. So we should think about what kind of min-SVN we actually want to report. If we report min-SVN as present in the firmware header, then that's a directive to firmware saying "fuses should be at least this", but the fuses might not have been blown by the time Caliptra runs, and Caliptra can't see what its new value is, as it doesn't manage fuses. Also if fuses are X and min-SVN is X-1, that's no less trustworthy a state than if min-SVN were X. So ignoring fuses seems a bit weird.

We could report what the fuse values were at boot-up. If min-SVN caused fuses to increment in the current boot cycle, a power cycle would be needed to actually get the attested evidence to update.

We could report the soft-SVN latched in the data vault. We _can't_ report that on FMC's alias key cert, though, because we don't re-generate that across hitless updates. Would need FMC to sample it and put it in RT FW's alias cert in order for it to reflect reality.

I'm not sure what all would be interesting to a verifier. @ericeilertson, any thoughts?

Reply:
Resolving this; we now use fuse-SVN as per https://github.com/chipsalliance/caliptra-sw/issues/269